src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,201 @@
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        final int numberToFetch = 50;
+        List<BibEntry> entryList = new LinkedList<>();
+
+        if (!query.isEmpty()) {
+            String searchTerm = replaceCommaWithAND(query);
+
+            // get the ids from entrez
+            SearchResult result = getIds(searchTerm, 0, 1);
+
+            if (result.count == 0) {
+                LOGGER.warn(Localization.lang("No references found"));
+                return Collections.emptyList();
+            }
+
+            for (int i = 0; i < numberToFetch; i++) {
+                // get the ids from entrez
+                result = getIds(searchTerm, i, numberToFetch - i);
+
+                List<BibEntry> bibs = fetchMedline(result.ids);
+                entryList.addAll(bibs);
+            }
+            return entryList;
+        }
+        throw new FetcherException(Localization.lang("Input error") +". " +Localization.lang("Please enter a comma separated list of Medline IDs (numbers) or search terms."));
+    }
+
+    /**
+     * Fetch and parse an medline item from eutils.ncbi.nlm.nih.gov.
+     *
+     * @param id One or several ids, separated by ","
+     * @return Will return an empty list on error.
+     */
+    private static List<BibEntry> fetchMedline(String id) {
+        try {
+            URIBuilder uriBuilder = new URIBuilder(API_MEDLINE_FETCH);
+            uriBuilder.addParameter("db", "pubmed");
+            uriBuilder.addParameter("retmode", "xml");
+            uriBuilder.addParameter("rettype", "citation");
+            uriBuilder.addParameter("id", id);
+            URL fetchURL = uriBuilder.build().toURL();
+            URLConnection data = fetchURL.openConnection();
+            ParserResult result = new MedlineImporter().importDatabase(
+                    new BufferedReader(new InputStreamReader(data.getInputStream(), StandardCharsets.UTF_8)));
+            if (result.hasWarnings()) {
+                LOGGER.warn(result.getErrorMessage());
+            }
+            return result.getDatabase().getEntries();
+        } catch (URISyntaxException | IOException e) {
+            LOGGER.warn(e.getLocalizedMessage(), e);
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        String cleanQuery = identifier.trim().replace(';', ',');
+
+        if (cleanQuery.matches(CLEAN_ID_QUERY)) {
+            List<BibEntry> entry = fetchMedline(cleanQuery);
+
+            if (entry.isEmpty()) {
+                LOGGER.warn(Localization.lang("No references found"));
+            }
+            LOGGER.info("Fetcher " + getName() + "found more than one result for identifier " + identifier
+                    + ". We will use the first entry.");
+            return Optional.of(entry.get(0));
+
+        }
+        return Optional.empty();
+    }
+
+    static class SearchResult {
```

Comment:
I don't really like this class, I would extract it to a new file and rename it to something like `MedlineSearchResult` or similar.